src/backend/src/bitcoin_utils.rs

```diff
@@ -0,0 +1,75 @@
+use std::str::FromStr;
+use ic_cdk::api::management_canister::bitcoin::{BitcoinNetwork};
+use k256::sha2;
+use k256::sha2::Digest;
+
+fn parse_derivation_path(path: &str) -> Vec<Vec<u8>> {
+    let re = regex::Regex::new(r"^m/44'(/\d+'?){2}/[01]/\d+$").unwrap();
+    assert!(re.is_match(path), "Invalid BIP-44 derivation path format");
+
+    path.split('/')
+        .skip(1) // Skip the leading 'm'
+        .map(|s| {
+            if s.ends_with('\'') {
+                // Handle hardened derivation (apostrophe indicates hardened)
+                // Value 0x80000000 is added to represent a hardened key.
+                // It is the hexadecimal of the smallest 32-bit value with the high-order bit set
+                // More details: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#Extended_keys
+                let index = u32::from_str(&s[..s.len() - 1]).unwrap() + 0x80000000;
+                index.to_be_bytes().to_vec()
+            } else {
+                let index = u32::from_str(s).unwrap();
+                index.to_be_bytes().to_vec()
+            }
+        })
+        .collect()
+}
+
+pub fn network_to_derivation_path(network: BitcoinNetwork) -> Vec<Vec<u8>> {
+    // This is a simplified version of the complete flow to create a derivation path.
+    // The documentation on the complete flow can be found at https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
+    // For the current implementation and to spare computation workload, we are using the default account 0, change 0, and address index 0.
+    // For the Regtest network, we are using the Testnet derivation path, even if it is not necessary to conform to BIP-44, since it is a local node.
+    // However, we do that because we would like to have the same behavior of the Testnet network.
+    let coin_type = match network {
+        BitcoinNetwork::Mainnet => "0",
+        BitcoinNetwork::Testnet => "1",
+        BitcoinNetwork::Regtest => "1",
+    };
+    let path = format!("m/44'/{}'/0'/0/0", coin_type);
+    parse_derivation_path(&path)
+}
+
+
+
+
+fn sha256(data: &[u8]) -> Vec<u8> {
+    let mut hasher = sha2::Sha256::new();
+    hasher.update(data);
+    hasher.finalize().to_vec()
+}
+fn ripemd160(data: &[u8]) -> Vec<u8> {
+    let mut hasher = ripemd::Ripemd160::new();
+    hasher.update(data);
+    hasher.finalize().to_vec()
+}
+
+// Converts a public key to a P2PKH address.
+pub fn public_key_to_p2pkh_address(public_key: &[u8], network: BitcoinNetwork) -> String {
```

Comment:
Is there a standard set of test vectors for this?  Or can we get a set of test cases from a fairly authoritative codebase?

Reply:
yes, indeed there is with [BIP-0038](https://github.com/bitcoin/bips/blob/master/bip-0038.mediawiki)

I will implement it in the tests

Reply:
Actually, @bitdivine , I have created a test that uses the `CALLER` canister and must match the output defined by `CALLER_BTC_ADDRESS`.

The test is quite similar to the ETH function with the same scope.

Do you think we still need to try it out on more examples?

Reply:
I would, if you can.  Where did you get the Bitcoin address msx83kFfWPib2YbnUk6xTpBn85vefcaVX8 from, used in the test?